src/software/thunderscope/gl/layers/gl_obstacle_layer.py

```diff
@@ -0,0 +1,107 @@
+            for primitive in primitive_set
+            if primitive.HasField("move")
+        ]
+
+        poly_obstacle_graphics_index = 0
+        circle_obstacle_graphics_index = 0
+
+        for obstacles in obstacles_ptrs:
+            for obstacle in obstacles:
+
+                for poly_obstacle in obstacle.polygon:
+
+                    # Get a previously cached graphic or create a new one
+                    if poly_obstacle_graphics_index >= len(self.poly_obstacle_graphics):
+                        poly_obstacle_graphic = GLLinePlotItem(
+                            color=Colors.NAVIGATOR_OBSTACLE_COLOR
+                        )
+                        self.poly_obstacle_graphics.append(poly_obstacle_graphic)
+                    else:
+                        poly_obstacle_graphic = self.poly_obstacle_graphics[
+                            poly_obstacle_graphics_index
+                        ]
+                    poly_obstacle_graphics_index += 1
+
+                    # In order to close the polygon, we need to include the first point at the end of
+                    # the list of points in the polygon
+                    polygon_points = (
+                        list(poly_obstacle.points) + poly_obstacle.points[:1]
+                    )
+
+                    poly_obstacle_graphic.setData(
+                        pos=np.array(
+                            [
+                                [point.x_meters, point.y_meters, 0]
+                                for point in polygon_points
+                            ]
+                        ),
+                    )
+
+                for circle_obstacle in obstacle.circle:
+
+                    # Get a previously cached graphic or create a new one
+                    if circle_obstacle_graphics_index >= len(
+                        self.circle_obstacle_graphics
+                    ):
+                        circle_obstacle_graphic = GLCircle(
+                            color=Colors.NAVIGATOR_OBSTACLE_COLOR
+                        )
+                        self.circle_obstacle_graphics.append(circle_obstacle_graphic)
+                    else:
+                        circle_obstacle_graphic = self.circle_obstacle_graphics[
+                            circle_obstacle_graphics_index
+                        ]
+                    circle_obstacle_graphics_index += 1
+
+                    circle_obstacle_graphic.set_radius(circle_obstacle.radius)
+                    circle_obstacle_graphic.set_position(
+                        circle_obstacle.origin.x_meters,
+                        circle_obstacle.origin.y_meters,
+                    )
+
+        # Remove graphics we don't need anymore
+        while poly_obstacle_graphics_index < len(self.poly_obstacle_graphics):
+            self.poly_obstacle_graphics.pop()
+        while circle_obstacle_graphics_index < len(self.circle_obstacle_graphics):
+            self.circle_obstacle_graphics.pop()
```

Comment:
Also, would it make sense to use `slice` if you were gonna keep this?

Reply:
you're right, we can use `_bring_list_to_length`. The way it's written right now is maybe a bit more efficient since we don't have to count the number of obstacle polygons/circles, but it isn't consistent with how other layers are structured